test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -11079,25 +11090,45 @@ def add_server_dyn_res(self, custom_resource, script_body=None,
         :param validate: if True (the default), validate the
                          configuration settings.
         :type validate: bool
+        :param dirname: the file will be created in this directory
+        :type dirname: str or None
+        :param host: the hostname on which dyn res script is created
+        :type host: str or None
+        :param perm: perm to use while creating scripts
+                     (must be octal like 0777)
+        :param prefix: the file name will begin with this prefix
+        :type prefix: str
+        :param suffix: the file name will end with this suffix
+        :type suffix: str
         """
         if res_file is not None:
-            f = open(file)
-            script_body = f.readlines()
-            f.close()
+            with open(res_file) as f:
+                script_body = f.readlines()
+                self.du.chmod(hostname=host, path=res_file, mode=perm,
+                              sudo=True)
         else:
-            res_file = self.du.create_temp_file(prefix='PtlPbsSchedConfig',
-                                                body=script_body)
+            if dirname is None:
+                dirname = self.pbs_conf['PBS_HOME']
+            tmp_file = self.du.create_temp_file(prefix=prefix, suffix=suffix,
+                                                body=script_body,
+                                                hostname=host)
+            res_file = os.path.join(dirname, tmp_file.split(os.path.sep)[-1])
+            self.du.run_copy(host, tmp_file, res_file, sudo=True,
+                             preserve_permission=False)
+            self.du.chown(hostname=host, path=res_file, uid=0, gid=0,
+                          sudo=True)
+            self.du.chmod(hostname=host, path=res_file, mode=perm, sudo=True)
+            if host is None:
+                self.dyn_created_files.append(res_file)
```

Comment:
i think DshUtuls's rm() can delete files on remote hosts as well

Reply:
You are right, but I'm not storing hostname along with the files. Although that can be done but since it is such a remote case for our tests, I just left it. Also, since the file is created inside PBS_HOME it gets cleaned up when the PBS is uninstalled in out testbed machines.